tests/test_data_adapter.py

```diff
@@ -301,6 +302,132 @@ def test_geodataset_set_nodata(artifact_data: DataCatalog):
     assert ds.vector.nodata == -99
 
 
+def test_dataset_get_data(timeseries_ds, tmpdir):
+    path = str(tmpdir.join("test.nc"))
+    timeseries_ds.to_netcdf(path)
+    dataset_adapter = DatasetAdapter(path=path, driver="netcdf")
+    ds1 = dataset_adapter.get_data()
+    assert isinstance(ds1, xr.Dataset)
+    assert ds1.identical(timeseries_ds)
+
+
+def test_dataset_to_file(timeseries_ds, tmpdir):
+    path = str(tmpdir.join("test1.nc"))
+    encoding = {k: {"zlib": True} for k in timeseries_ds.vector.vars}
+    timeseries_ds.to_netcdf(path, encoding=encoding)
+    dataset_adapter = DatasetAdapter(path=path, driver="netcdf")
+    fn_out, driver = dataset_adapter.to_file(
+        data_root=tmpdir, data_name="test2", driver="netcdf"
+    )
+    assert driver == "netcdf"
+    assert fn_out == str(tmpdir.join("test2.nc"))
+    ds2 = xr.open_dataset(fn_out)
+    assert ds2.identical(timeseries_ds)
+
+    variables = ["col1", "col2"]
+    fn_out, driver = dataset_adapter.to_file(
+        data_root=tmpdir, data_name="test3", driver="netcdf", variables=variables
+    )
+    for variable in variables:
+        ds3 = xr.open_dataset(fn_out.format(variable=variable))
+        assert variable in ds3.vector.vars
+
+    fn_out, driver = dataset_adapter.to_file(
+        data_root=tmpdir, data_name="test4", driver="zarr"
+    )
+
+    ds_zarr = xr.open_zarr(fn_out)
+    assert ds_zarr.identical(timeseries_ds)
+
+    with pytest.raises(ValueError, match="Dataset: Driver fake-driver unknown."):
+        dataset_adapter.to_file(
+            data_root=tmpdir, data_name="test4", driver="fake-driver"
+        )
+
+
+def test_dataset_read_data(tmpdir, timeseries_ds):
+    zarr_path = str(tmpdir.join("zarr_data"))
+    timeseries_ds.to_zarr(zarr_path)
+    dataset_adapter = DatasetAdapter(path=zarr_path, driver="zarr")
+    dataset_adapter.get_data(variables=["col1", "col2"])
+
+    dataset_adapter = DatasetAdapter(path=zarr_path, driver="fake-driver")
+    with pytest.raises(ValueError, match=r"Dataset: Driver fake-driver unknown"):
+        dataset_adapter.get_data(variables=["col1", "col2"])
+
+
+def test_dataset_set_nodata(tmpdir, timeseries_ds):
+    path = str(tmpdir.join("test.nc"))
+    timeseries_ds.to_netcdf(path)
+
+    nodata = -999
```

Comment:
out of curiosity, does it also accept float nodata?

Reply:
yes, you can set any no data value